src/Common/src/System/Net/Logging/GlobalLog.cs

```diff
@@ -136,7 +134,10 @@ internal static void SetThreadSource(ThreadKinds source)
             {
                 EventSourceLogging.Log.WarningMessage("The stack has been corrupted.");
                 ThreadKinds last = ThreadKindStack.Pop() & ThreadKinds.SourceMask;
-                Assert(last == source || last == ThreadKinds.Other, "Thread source changed.|Was:({0}) Now:({1})", last, source);
+                if (last != source && last != ThreadKinds.Other)
```

Comment:
I can.  I didn't because this code is only ever used in debug, but for consistency I'll put it in.